src/rules/noInvalidThisRule.ts

```diff
@@ -59,39 +59,64 @@ export class Rule extends Lint.Rules.AbstractRule {
 
 function walk(ctx: Lint.WalkContext<boolean>): void {
     const { sourceFile, options: checkFuncInMethod } = ctx;
+
+    const enum parentType {
+        None,
+        Class,
+        ClassMethod,
+        BoundRegularFunction,
+        UnboundRegularFunction,
+    }
+
+    function thisIsAllowed(parent: parentType): boolean {
```

Comment:
Nit: instead of this three-line function, how about `const thisAllowedTypes = new Set([ParentType.ClassMethod, ParentType.BoundRegularFunction]);`, and use its `.has`?